### PR TITLE
Added getInteractionObject() to model

### PR DESCRIPTION
**Pull request by brian-learningpool**
Also moved getResponseType() and getResponse() functions from view to model.

### CODE REVIEW

File: js/mcqModel.js [hunk 1]
```diff
@@ -168,10 +168,59 @@ define([
...
+        getInteractionObject: function() {
+            var interactions = {
+                correctResponsesPattern: [],
+                choices: []
+            };
+
+            interactions.choices = this.get('_items').map(function(item) {
+                return {
+                    id: (item._index + 1).toString(),
+                    description: item.text
```
Comment (brian-learningpool) [inline js/mcqModel.js]: I'm just wondering if we should return the text mapped to the active language here?

Comment (oliverfoster) [inline js/mcqModel.js]: that's global to the course, i'm not sure it's necessary to include that with every call?

Comment (oliverfoster) [inline js/mcqModel.js]: i always get stuck with where the line get's drawn between xapi api and adapt provision.

Comment (barry-learningpool) [inline js/mcqModel.js]: @brian-learningpool I think that's a good idea, seems to be the appropriate format for xAPI

Comment (oliverfoster) [inline js/mcqModel.js]: i don't think we're trying to implement xapi throughout the course, in the same way we didn't implement scorm. we just need to provide apis for the appropriate information from each plugin. passing the language with every interaction.choices seems overkill, especially as the xapi plugin can derive that itself.

Comment (brian-learningpool) [inline js/mcqModel.js]: Hi @oliverfoster, I actually have the xAPI plugin handling this, i.e. where it sees a `choices` object it checks if the description contains a string or an object.  If it's a string it maps it accordingly.

I was actually on the fence about it, but if it's acceptable as is -- i.e. plugin neutral -- I'm ok with it.

Comment (oliverfoster) [inline js/mcqModel.js]: looks lovely 👍

**Comment by oliverfoster** [comment]
Yup it will. It doesn't look like there is an outstanding pr @moloko

**Comment by brian-learningpool** [comment]
@moloko , @oliverfoster, should we create a corresponding issue to update Spoor to check those functions on the model rather than the view for v3?

**Comment by oliverfoster** [comment]
yes @brian-learningpool good plan

**Comment by brian-learningpool** [comment]
The corresponding PR on Spoor is here: https://github.com/adaptlearning/adapt-contrib-spoor/pull/171

**Pull request merged**